overlord/ifacestate/helpers.go

```diff
@@ -580,12 +589,18 @@ func (c *autoConnectChecker) addAutoConnections(newconns map[string]*interfaces.
 		// providing the same interface. In that situation we
 		// want to ignore any candidates in ubuntu-core and
 		// simply go with those from the new core snap.
-		candSlots = filterUbuntuCoreSlots(candSlots)
+		candSlots, arities = filterUbuntuCoreSlots(candSlots, arities)
 
 		applicable := candSlots
 		// candidate arity check
-		if len(candSlots) != 1 {
-			applicable = nil
+		for _, arity := range arities {
+			if !arity.SlotsPerPlugAny() {
+				// ATM not any (*) => none or exactly one
```

Comment:
This comment could be expanded a little bit, it's very terse right now. I know now what it means (and it's nicely tested) but I'm not sure if I will know in a week :) But not a blocker, just a nitpick.